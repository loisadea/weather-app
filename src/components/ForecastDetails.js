import React from "react";

import "../styles/ForecastDetails.css";

function ForecastDetails({ date, temperature, wind, humidity }) {
  const dateHuman = new Date(date).toLocaleDateString("en-UK", {
    weekday: "short",
    year: "numeric",
    month: "short",
    day: "numeric",
  });
  return (
    <div className="ForecastDetails">
      <div className="ForecastDetails__date">{`${dateHuman}`}</div>
      <div className="ForecastDetails__min_temperature">
        <span className="ForecastDetails__min_temperature__label">
          Min temperature:
        </span>
        {`  ${temperature.min} °C`}
      </div>
      <div className="ForecastDetails__max_temperature">
        <span className="ForecastDetails__max_temperature__label">
          Max temperature:
        </span>
        {`  ${temperature.max} °C`}
      </div>
      <div className="ForecastDetails__humidity">
        <span className="ForecastDetails__humidity__label">Humidity:</span>
        {`  ${humidity}%`}
      </div>
      <div className="ForecastDetails__wind">
        <span className="ForecastDetails__wind__label">Wind:</span>
        {`  ${wind.speed}mph`}
      </div>
    </div>
  );
}

export default ForecastDetails;
