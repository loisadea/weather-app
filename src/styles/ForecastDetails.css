div.ForecastDetails {
    text-align: center;
    margin: 10px;
    font-size: 14px;
}

div.ForecastDetails__date,
div.ForecastDetails__min_temperature,
div.ForecastDetails__max_temperature,
div.ForecastDetails__humidity,
div.ForecastDetails__wind {
    margin: 5px;
}


div.ForecastDetails__date {
    font-size: 20px;
    font-weight: bold;
}

span.ForecastDetails__min_temperature__label,
span.ForecastDetails__max_temperature__label,
span.ForecastDetails__humidity__label,
span.ForecastDetails__wind__label {
    font-weight: bold;
}