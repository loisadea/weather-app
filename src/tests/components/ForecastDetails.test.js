import React from "react";

import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const fakeForecast = {
    date: 1692024554000,
    temperature: {
      max: 11,
      min: 4,
    },
    wind: {
      speed: 13,
      direction: "s",
    },
    humidity: 30,
    description: "Clear",
    icon: "800",
  };
  const fakeForecastComponent = (
    <ForecastDetails
      date={fakeForecast.date}
      temperature={fakeForecast.temperature}
      wind={fakeForecast.wind}
      humidity={fakeForecast.humidity}
    />
  );

  it("renders correctly", () => {
    const { asFragment } = render(fakeForecastComponent);
    expect(asFragment()).toMatchSnapshot();
  });

  it("shows the date", () => {
    const { getByText } = render(fakeForecastComponent);
    expect(getByText("Mon, 14 Aug 2023")).toHaveClass("ForecastDetails__date");
  });

  it("shows the min temperature", () => {
    const { getByText } = render(fakeForecastComponent);
    expect(getByText(/Min temperature:/)).toBeInTheDocument();
    expect(getByText(/4 °C/)).toBeInTheDocument();
  });

  it("shows the max temperature", () => {
    const { getByText } = render(fakeForecastComponent);
    expect(getByText(/Max temperature:/)).toBeInTheDocument();
    expect(getByText(/11 °C/)).toBeInTheDocument();
  });

  it("shows the humidty", () => {
    const { getByText } = render(fakeForecastComponent);
    expect(getByText(/Humidity:/)).toBeInTheDocument();
    expect(getByText(/30%/)).toBeInTheDocument();
  });

  it("shows the wind", () => {
    const { getByText } = render(fakeForecastComponent);
    expect(getByText(/Wind:/)).toBeInTheDocument();
    expect(getByText(/13mph/)).toBeInTheDocument();
  });
});
