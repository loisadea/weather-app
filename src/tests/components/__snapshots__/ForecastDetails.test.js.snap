// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ForecastDetails"
  >
    <div
      class="ForecastDetails__date"
    >
      Mon, 14 Aug 2023
    </div>
    <div
      class="ForecastDetails__min_temperature"
    >
      <span
        class="ForecastDetails__min_temperature__label"
      >
        Min temperature:
      </span>
        4 °C
    </div>
    <div
      class="ForecastDetails__max_temperature"
    >
      <span
        class="ForecastDetails__max_temperature__label"
      >
        Max temperature:
      </span>
        11 °C
    </div>
    <div
      class="ForecastDetails__humidity"
    >
      <span
        class="ForecastDetails__humidity__label"
      >
        Humidity:
      </span>
        30%
    </div>
    <div
      class="ForecastDetails__wind"
    >
      <span
        class="ForecastDetails__wind__label"
      >
        Wind:
      </span>
        13mph
    </div>
  </div>
</DocumentFragment>
`;
